File: get_r_in_Wn.m
%****************************************************************************
%     Code writer     : EOF
%     Code file       : bit_reverse.m
%     Code date       : 2014.09.04
%     e-mail          : jasonleaster@gmail.com
%
%     Function        : ret = get_r_in_Wn(k,m,N,bits)
%
%     Input Parameter : @k, the location of input signal
%                       @m, the current layyer
%                       @N, the total number of inputed signal
%                       @bits, how many bits should be used to represent 'N'
%
%     Output Parameter: @ret , the value of 'r'
%             
%****************************************************************************

function ret = get_r_in_Wn(k,m,N,bits)

         tmp = bitshift(k,bits - m);
         
         ret = bitand(tmp,(2^m) - 1);    
end

File: FFT_in_C/get_r_in_Wn.c
/******************************************************************************
code writer	:	EOF
code file	:	get_r_in_Wn.c
code date	:	2014.09.17
e-mail		:	jasonleaster@gmail.com

	Input Parameter : @k, the location of input signal
                          @m, the current layyer
                          @N, the total number of inputed signal
                          @bits, how many bits should be used to represent 'N'

	Output Parameter: @ret , the value of 'r'
*******************************************************************************/
int get_r_in_Wn(int k, int m, int bits)
{
	int tmp = k<<(bits-m);

	return tmp&((1<<m) -1);
}
